honeybee_grasshopper_energy/src/HB ConstructionSet Interior.py

```diff
@@ -0,0 +1,117 @@
+-
+
+    Args:
+        _constr_set: A ConstructionSet object for which the interior constructions
+            will be edited.
+        _interior_wall_: A construction object for interior walls (or text for
+            the name of the construction within the library).
+        _ceiling_: A construction object for ceilings (or text for the name of
+            the construction within the library).
+        _interior_floor_: A construction object for interior floors (or text for
+            the name of the construction within the library).
+        _interior_window_: A construction object for all apertures with a Surface
+            boundary condition. This can also be text for the name of the
+            construction within the library.
+        _interior_door_: A construction object for all opaque doors with a Surface
+            boundary condition. This can also be text for the name of the
+            construction within the library.
+        _int_glass_door_: A construction object for all glass doors with a Surface
+            boundary condition. This can also be text for the name of the
+            construction within the library.
+    
+    Returns:
+        constr_set: A ConstructionSet object that can be assigned to Honeybee
+            Rooms in order to specify all default constructions on the Room.
+"""
+
+ghenv.Component.Name = "HB ConstructionSet Interior"
+ghenv.Component.NickName = 'ConstrSetInterior'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Energy"
+ghenv.Component.SubCategory = '0 :: Basic Properties'
+ghenv.Component.AdditionalHelpFromDocStrings = "3"
+
+try:
+    from honeybee_energy.constructionset import ConstructionSet
+    from honeybee_energy.lib.constructionsets import construction_set_by_name
+    from honeybee_energy.lib.constructions import opaque_construction_by_name, \
+        window_construction_by_name
+    from ladybug_rhino.grasshopper import all_required_inputs, give_warning
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee:\n\t{}'.format(e))
+
+
+def opaque_constr(construction):
+    """Get an OpaqueConstrucion from the library if it's a string."""
+    if isinstance(construction, str):
+        return opaque_construction_by_name(construction)
+    return construction
+
+
+def window_constr(construction):
+    """Get an WindowConstrucion from the library if it's a string."""
+    if isinstance(construction, str):
+        return window_construction_by_name(construction)
+    return construction
+
+
+def give_symmetric_warning(constr_type, construction):
+    """Give a warning on the component that a construction is asymmetric."""
+    give_warning(
+        ghenv.Component, 'Input {} is asymmetric (materials in reversed order ' \
+        'do not equal those in the current order).\nThis can cause issues if the ' \
+        'resulting constr_set is applied across multiple Rooms.\nMaterials: {}'.format(
```

Comment:
Let's say that I make a ConstructionSet for a wooden house that I plan to apply to all of the Rooms in the model. If I create an asymmetric interior wall construction (ie. ['Plywood', 'Wood Stud', 'Solid Wood']) and then I apply it to all the Rooms.  Now, all of the interior walls will not have  constructions in reverse order from their adjacent walls, causing E+ to raise an exception and exit.

There is a way around this, which is why I only raise a warning here and not an exception.  You could create two interior ConstrucitonSets that have reversed materials from one another on interior walls and assign each to a different adjacent room. However, at that point, it's cleaner to just use the `ep_constr_` input on the "HB Solve Adjacency" component to take care of this rather than creating custom interior construction sets.  So that's why I feel a warning should be raised in this case.

Reply:
Thank you for the explanation. I'm not sure if I understand the difference between the two but it should become obvious once I use the components.